spec/ddtrace/contrib/rspec/instrumentation_spec.rb

```diff
@@ -5,36 +5,35 @@
 require 'ddtrace'
 
 RSpec.describe 'RSpec hooks' do
-  extend ConfigurationHelpers
-
   let(:configuration_options) { {} }
 
-  around do |example|
-    old_configuration = ::RSpec.configuration
-    ::RSpec.configuration = ::RSpec::Core::Configuration.new
-
+  before do
     Datadog.configure do |c|
       c.use :rspec, configuration_options
     end
-    example.run
+  end
+
+  # Yields to a block in a new RSpec global context. All RSpec
+  # test configuration and execution should be wrapped in this method.
+  def with_new_rspec_environment
+    old_configuration = ::RSpec.configuration
+    old_world = ::RSpec.world
+    ::RSpec.configuration = ::RSpec::Core::Configuration.new
+    ::RSpec.world = ::RSpec::Core::World.new
 
-    RSpec.configuration = old_configuration
-    Datadog.configuration.reset!
+    yield
+  ensure
+    ::RSpec.configuration = old_configuration
+    ::RSpec.world = old_world
   end
 
   it 'creates span for example' do
-    spans = []
-    spec = RSpec.describe 'some test' do
-      let(:active_span) { Datadog.configuration[:rspec][:tracer].active_span }
-
-      it 'foo' do
-        spans << active_span
-      end
+    spec = with_new_rspec_environment do
+      RSpec.describe 'some test' do
+        it 'foo' do; end
+      end.tap(&:run)
```

Comment:
Minor -- we seem to be attaching `run` to every call to `with_new_rspec_environment`. What do you think of automatically calling `run` on the result of `yield` inside the helper method?

Reply:
I thought of doing that, but chose against it as it gives a seemly minor code simplification at the expense of removing flexibility from `with_new_rspec_environment` users.

Having `with_new_rspec_environment` solely "give you a clean rspec environment" made more sense to me while writing these tests.